src/m365/spo/commands/applicationcustomizer/applicationcustomizer-get.spec.ts

```diff
@@ -0,0 +1,534 @@
+        webUrl: webUrl,
+        scope: 'Web'
+      }
+    }), new CommandError(`Multiple application customizers with title '${title}' found. Please disambiguate using IDs: a70d8013-3b9f-4601-93a5-0e453ab9a1f3, 63aa745f-b4dd-4055-a4d7-d9032a0cfc59`));
+  });
+
+  it('handles error when multiple application customizers with the specified clientSideComponentId found', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${webUrl}/_api/Web/UserCustomActions?$filter=ClientSideComponentId eq guid'7096cded-b83d-4eab-96f0-df477ed7c0bc' and Location eq 'ClientSideExtension.ApplicationCustomizer'`) {
+        return {
+          value: [
+            {
+              ClientSideComponentId: clientSideComponentId,
+              ClientSideComponentProperties: "'{testMessage:Test message}'",
+              CommandUIExtension: null,
+              Description: null,
+              Group: null,
+              HostProperties: '',
+              Id: 'a70d8013-3b9f-4601-93a5-0e453ab9a1f3',
+              ImageUrl: null,
+              Location: 'ClientSideExtension.ApplicationCustomizer',
+              Name: 'YourName',
+              RegistrationId: null,
+              RegistrationType: 0,
+              Rights: [Object],
+              Scope: 3,
+              ScriptBlock: null,
+              ScriptSrc: null,
+              Sequence: 0,
+              Title: 'YourAppCustomizer 1',
+              Url: null,
+              VersionOfUserCustomAction: '16.0.1.0'
+            },
+            {
+              ClientSideComponentId: clientSideComponentId,
+              ClientSideComponentProperties: "'{testMessage:Test message}'",
+              CommandUIExtension: null,
+              Description: null,
+              Group: null,
+              HostProperties: '',
+              Id: '63aa745f-b4dd-4055-a4d7-d9032a0cfc59',
+              ImageUrl: null,
+              Location: 'ClientSideExtension.ApplicationCustomizer',
+              Name: 'YourName',
+              RegistrationId: null,
+              RegistrationType: 0,
+              Rights: [Object],
+              Scope: 3,
+              ScriptBlock: null,
+              ScriptSrc: null,
+              Sequence: 0,
+              Title: 'YourAppCustomizer 2',
+              Url: null,
+              VersionOfUserCustomAction: '16.0.1.0'
+            }
+          ]
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.rejects(command.action(logger, {
+      options: {
+        clientSideComponentId: clientSideComponentId,
+        webUrl: webUrl,
+        scope: 'Web'
+      }
+    }), new CommandError(`Multiple application customizers with Client Side Component Id '${clientSideComponentId}' found. Please disambiguate using IDs: a70d8013-3b9f-4601-93a5-0e453ab9a1f3, 63aa745f-b4dd-4055-a4d7-d9032a0cfc59`));
+  });
+
+  it('handles error when found custom action by id is not an Application Customizer', async () => {
```

Comment:
I think this description does not describe what we actually check. First of all, we already filter out any other custom action which is not an application customizer. Secondly here we check if we find a application customizer by id 🤔

Reply:
Thank you @Adam-it for pointing this.
As `spo.getCustomActionById` does not support filter, we might get custom action that is not ApplicationCustomizer.

Renamed the test case as `handles error when no valid application customizer with the specified id found`